### PR TITLE
Support constructing `cudf.Scalar` objects from host side lists

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Makes it so that this works:

```
>>> slr = cudf.Scalar([1,2,3])
>>> slr
Scalar([1, 2, 3], dtype=list)
>>> slr.dtype
ListDtype(int64)
```

Required for `ListColumn.__setitem__`

### CODE REVIEW

**Review by galipremsagar — APPROVED** [review]

**Comment by galipremsagar** [comment]
rerun tests

**Review by isVoid — APPROVED** [review]
Would love to revisit: https://github.com/rapidsai/cudf/blob/d30695c21ac551e0039a94c4270798b7bf57b1e3/python/cudf/cudf/tests/test_groupby.py#L1788-L1789
after this is merged. (Can't remember what's the exact issue)

**Comment by brandon-b-miller** [comment]
> Would love to revisit:
> 
> https://github.com/rapidsai/cudf/blob/d30695c21ac551e0039a94c4270798b7bf57b1e3/python/cudf/cudf/tests/test_groupby.py#L1788-L1789
> 
> 
> after this is merged. (Can't remember what's the exact issue)

I think it has to do with null filling, it probably fails to construct here using a listdtype:

https://github.com/rapidsai/cudf/blob/d30695c21ac551e0039a94c4270798b7bf57b1e3/python/cudf/cudf/_lib/copying.pyx#L660

As an aside, thank you for pointing that out as this case is actually totally unhandled in this PR - I think it makes sense to go back and add it here before merging.

**Comment by brandon-b-miller** [comment]
@isVoid could you request changes on this PR to temporarily block merging?

**Review by isVoid — CHANGES_REQUESTED** [review]
Request change per above.

**Review by marlenezw — APPROVED** [review]

File: python/cudf/cudf/tests/test_list.py
Comment (brandon-b-miller) [reply python/cudf/cudf/tests/test_list.py:346]: Before https://github.com/rapidsai/cudf/pull/8442, if we wanted to make a series with an `NA` in it out of a list, we had to pass `None`. This is because the list ends up getting converted into an arrow array and then we know how to make a series or other gpu object out of that. But it didn't recognize our own `cudf.NA` object and would crash. This meant we needed to pass `None` to indicate `NA` and then after rountripping, test that there was `NA` in the same places there was `None` originally. After that PR though, we can rountrip `NA` correctly.

**Comment by brandon-b-miller** [comment]
This should be ready for another look.

File: python/cudf/cudf/core/column/column.py
Comment (isVoid) [inline python/cudf/cudf/core/column/column.py:593]:
```suggestion
```

**Review by isVoid — APPROVED** [review]
lgtm, thanks for addressing above.

**Comment by brandon-b-miller** [comment]
Discovered a bug while working forward on `__setitem__` should be resolved by the latest commit.

**Comment by brandon-b-miller** [comment]
rerun tests

**Comment by brandon-b-miller** [comment]
rerun tests

**Comment by brandon-b-miller** [comment]
@gpucibot merge

**Pull request merged**
